spec/models/spree/gateway/stripe_sca_spec.rb

```diff
@@ -3,80 +3,117 @@
 require 'spec_helper'
 
 describe Spree::Gateway::StripeSCA, type: :model do
-  before { Stripe.api_key = "sk_test_12345" }
+  let(:secret) { ENV.fetch('STRIPE_SECRET_TEST_API_KEY', nil) }
+
+  let(:order) { create(:order_ready_for_payment) }
+
+  let(:year_valid) { Time.zone.now.year.next }
+
+  let(:credit_card) { create(:credit_card, gateway_payment_profile_id: pm_card.id) }
 
-  let(:order) { create(:order_with_totals_and_distribution) }
-  let(:credit_card) { create(:credit_card) }
   let(:payment) {
     create(
       :payment,
-      state: "checkout",
       order:,
       amount: order.total,
       payment_method: subject,
       source: credit_card,
-      response_code: "12345"
+      response_code: payment_intent.id
     )
   }
+
   let(:gateway_options) {
     { order_id: order.number }
   }
-  let(:payment_authorised) {
-    payment_intent(payment.amount, "requires_capture")
-  }
-  let(:capture_successful) {
-    payment_intent(payment.amount, "succeeded")
-  }
 
-  describe "#purchase" do
-    it "captures the payment" do
-      stub_request(:get, "https://api.stripe.com/v1/payment_intents/12345").
-        to_return(status: 200, body: payment_authorised)
-      stub_request(:post, "https://api.stripe.com/v1/payment_intents/12345/capture").
-        with(body: { "amount_to_capture" => order.total }).
-        to_return(status: 200, body: capture_successful)
+  describe "#purchase", :vcr, :stripe_version do
+    before { Stripe.api_key = secret }
+
+    let!(:pm_card) do
+      Stripe::PaymentMethod.create({
+                                     type: 'card',
+                                     card: {
+                                       number: '4242424242424242',
+                                       exp_month: 12,
+                                       exp_year: year_valid,
+                                       cvc: '314',
+                                     },
+                                   })
+    end
+    let!(:payment_intent) do
+      Stripe::PaymentIntent.create({
+                                     amount: 1000, # given in AUD cents
+                                     currency: 'aud', # AUD to match order currency
+                                     payment_method: pm_card,
+                                     payment_method_types: ['card'],
+                                     capture_method: 'manual',
+                                   })
+    end
+
+    # Stripe acepts amounts as positive integers representing how much to charge
+    # in the smallest currency unit
+    let(:capture_amount) { order.total.to_i * 10 }
+    let(:response) { subject.purchase(capture_amount, credit_card, gateway_options) }
```

Comment:
I had to extract the response to a let declaration, so that I can use within the `expect{}` group below (calls Stripe payment intent status change), as well as outside of it (asserts on the Active Merchant response)

Reply:
I can remove this now, as discussed below.